pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -744,3 +747,111 @@ func TestOnDNSResponse(t *testing.T) {
 		})
 	}
 }
+func TestFQDNCacheMinTTL(t *testing.T) {
+	currentTime := time.Now()
+	testIP := "192.168.1.1"
+	testFQDN := "fqdn-test-pod.lfx.test"
+	tests := []struct {
+		name            string
+		expectedTTL     time.Time
+		fqdnCacheMinTTL uint32
+		dnsMsg          *dns.Msg
+	}{
+		{
+			name:            "Response TTL less than FQDNCacheTTL",
+			expectedTTL:     currentTime.Add(10 * time.Second),
+			fqdnCacheMinTTL: 10,
+			dnsMsg: &dns.Msg{
+				Question: []dns.Question{
+					{Name: testFQDN, Qtype: dns.TypeA, Qclass: dns.ClassINET},
+				},
+				Answer: []dns.RR{
+					&dns.A{
+						Hdr: dns.RR_Header{
+							Name:   testFQDN,
+							Rrtype: dns.TypeA,
+							Class:  dns.ClassINET,
+							Ttl:    5,
+						},
+						A: net.ParseIP(testIP),
+					},
+				},
+			},
+		},
+		{
+			name:            "Response TTL more than FQDNCacheTTL",
+			expectedTTL:     currentTime.Add(10 * time.Second),
+			fqdnCacheMinTTL: 5,
+			dnsMsg: &dns.Msg{
+				Question: []dns.Question{
+					{Name: testFQDN, Qtype: dns.TypeA, Qclass: dns.ClassINET},
+				},
+				Answer: []dns.RR{
+					&dns.A{
+						Hdr: dns.RR_Header{
+							Name:   testFQDN,
+							Rrtype: dns.TypeA,
+							Class:  dns.ClassINET,
+							Ttl:    10,
+						},
+						A: net.ParseIP(testIP),
+					},
+				},
+			},
+		},
+		{
+			name:            "Response TTL equal to FQDNCacheTTL",
+			expectedTTL:     currentTime.Add(5 * time.Second),
+			fqdnCacheMinTTL: 5,
+			dnsMsg: &dns.Msg{
+				Question: []dns.Question{
+					{Name: testFQDN, Qtype: dns.TypeA, Qclass: dns.ClassINET},
+				},
+				Answer: []dns.RR{
+					&dns.A{
+						Hdr: dns.RR_Header{
+							Name:   testFQDN,
+							Rrtype: dns.TypeA,
+							Class:  dns.ClassINET,
+							Ttl:    5,
+						},
+						A: net.ParseIP(testIP),
+					},
+				},
+			},
+		},
+		{
+			name:            "FQDNCacheTTL is not set",
+			expectedTTL:     currentTime.Add(5 * time.Second),
+			fqdnCacheMinTTL: 0,
+			dnsMsg: &dns.Msg{
+				Question: []dns.Question{
+					{Name: testFQDN, Qtype: dns.TypeA, Qclass: dns.ClassINET},
+				},
+				Answer: []dns.RR{
+					&dns.A{
+						Hdr: dns.RR_Header{
+							Name:   testFQDN,
+							Rrtype: dns.TypeA,
+							Class:  dns.ClassINET,
+							Ttl:    5,
+						},
+						A: net.ParseIP(testIP),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeClock := newFakeClock(currentTime)
+			controller := gomock.NewController(t)
+			f, _ := newMockFQDNController(t, controller, nil, fakeClock, tc.fqdnCacheMinTTL)
+			require.Zero(t, fakeClock.TimersAdded())
```

Comment:
is this useful in the context of this test? Please add a short comment if you think it is necessary.